src/dotty/tools/dotc/core/SymDenotations.scala

```diff
@@ -1060,6 +1060,12 @@ object SymDenotations {
       else defn.RootClass
     }
 
+    /** Is this denotation public ? */
+    def isPublic(implicit ctx: Context): Boolean =
+      // private-qualified denotations do not have the Private flag set,
+      // see `ParserCommon#normalize`
+      !this.is(Protected | Private) && !privateWithin.exists
```

Comment:
Why would you need to check for `privateWithin.exists` if neither `Protected`  nor `Private` flag is set?